File: leetcode/2960-count-tested-devices-after-test-operations/countTestedDevices.ts
function countTestedDevices(batteryPercentages: number[]): number {
  let count = 0; // 已检测的数量 也代表 检测设备需要扣除的电量

  for (const percentage of batteryPercentages) {
    if (percentage > count) {
      count++;
    }
  }

  return count;
}

export default countTestedDevices;

File: leetcode/2960-count-tested-devices-after-test-operations/countTestedDevices.rs
use crate::Solution;

impl Solution {
    pub fn count_tested_devices(battery_percentages: Vec<i32>) -> i32 {
        let mut count = 0;

        for percentage in battery_percentages {
            if percentage > count {
                count = count + 1
            }
        }

        return count;
    }
}
